.github/workflows/formatting.yml

```diff
@@ -0,0 +1,43 @@
+name: Check formatting
+
+concurrency:
+  # Don't cancel on master, creating a PR when a push workflow is already going will cancel the push workflow in favour of the PR workflow
+  group: ${{ github.workflow }}-${{ github.ref == 'refs/heads/master' && github.run_id || github.event.number && github.head_ref || github.ref_name }}
```

Comment:
This is admittedly a bit difficult to read, I think some parentheses would make the intent clearer.

Reply:
It's identical to other parts of the CI, and identical to what it was before.